unit/util/string_utils/split_string.cpp

```diff
@@ -101,6 +101,39 @@ SCENARIO("split_string", "[core][utils][string_utils][split_string]")
 
     run_on_all_variants(string, delimiter, expected_results);
   }
+  GIVEN("An empty string")
+  {
+    std::string string = "";
+    WHEN("Splitting it")
+    {
+      expected_resultst expected_results;
+      expected_results.no_strip_no_remove = {""};
+      expected_results.strip_no_remove = {""};
+
+      // TODO(tkiley): This is probably wrong, since I'd expect removing empty
+      // TODO(tkiley): elements to return an empty vector here.
```

Comment:
I agree, we could change this to return an empty vector. I've looked at the places where `split_string()` is currently used, and they would also work when returning an empty vector (except an assertion `assert(!result.empty())` would need to be removed in `util/unwind.cpp`). Also once changed, the current usages of `split_string()` can be slightly simplified as some of them have a check for this case.